app/tests/test_middlewares.py

```diff
@@ -0,0 +1,83 @@
+import json
+from unittest.mock import MagicMock
+
+import pytest
+from flask import current_app
+from flask.testing import FlaskClient
+
+from app.main.middlewares.log_page_view import log_page_view
+
+
+@pytest.mark.parametrize(
+    "route_path, method, session_object, route_function, expected_response, expected_log_data",
+    [
+        (
+            "/test_route",
+            "GET",
+            {"user_id": "test_user"},
+            lambda: "Test Response",
+            b"Test Response",
+            {
+                "event": "page_view",
+                "user_id": "test_user",
+                "route": "/test_route",
+                "method": "GET",
+            },
+        ),
+        (
+            "/anonymous_route",
+            "GET",
+            {},
+            lambda: "Anonymous Response",
+            b"Anonymous Response",
+            {
+                "event": "page_view",
+                "user_id": "anonymous",
+                "route": "/anonymous_route",
+                "method": "GET",
+            },
+        ),
+        (
+            "/post_route",
+            "POST",
+            {"user_id": "test_user"},
+            lambda: "Post Response",
+            b"Post Response",
+            {
+                "event": "page_view",
+                "user_id": "test_user",
+                "route": "/post_route",
+                "method": "POST",
+            },
+        ),
+    ],
+)
+def test_log_page_view(
+    app,
+    client: FlaskClient,
+    route_path,
+    method,
+    session_object,
+    route_function,
+    expected_response,
+    expected_log_data,
+):
+    mock_logger = MagicMock()
```

Comment:
we don't need to mock to assert the logged output.

We can use the caplog fixture as described in the pytyest docs here: https://docs.pytest.org/en/latest/how-to/logging.html#caplog-fixture

Reply:
definitely, I've replaced with caplog - seems like we also do this elsewhere so we have a fixture setup